src/app/components/Transaction/index.tsx

```diff
@@ -156,10 +157,24 @@ export function Transaction(props: TransactionProps) {
     },
   }
 
-  const matchingConfiguration = transactionDictionnary[transaction.type as TransactionType][side]
-  const icon = matchingConfiguration.icon()
-  const header = matchingConfiguration.header()
-  const designation = matchingConfiguration.designation
+  let icon, header, designation
```

Comment:
These then have type `any` :/

Alternative:
```ts
  const unrecognizedTransaction: TransactionDictionary[TransactionType][TransactionSide] = {
    designation: t('account.otherTransaction.designation', 'Other address'),
    icon: () => <New />,
    header: () => (
      <Trans
        i18nKey="account.otherTransaction.header"
        t={t}
        components={[transaction.type]}
        defaults="Unrecognized transaction, type '<0></0>'"
      />
    ),
  }

  const isTypeRecognized = (type: string | undefined): type is TransactionType =>
    type ? type in transactionDictionary : false

  const matchingConfiguration = isTypeRecognized(transaction.type)
    ? transactionDictionary[transaction.type][side]
    : unrecognizedTransaction

  const icon = matchingConfiguration.icon()
  const header = matchingConfiguration.header()
  const designation = matchingConfiguration.designation
```

Reply:
hm, `isTypeRecognized`

Reply:
thanks, I'm fine with that. I'll add this in a commit